### PR TITLE
Team submission

### PR DESCRIPTION
**Pull request by cbare**
SYNPY-221

### CODE REVIEW

File: synapseclient/__main__.py [hunk 1]
```diff
@@ -545,28 +537,30 @@ def build_parser():
...
             help='Evaluation Name where the entity/file will be submitted')
     parser_submit.add_argument('--evaluation', type=str,
             help=argparse.SUPPRESS)  #mainly to maintain the backward compatibility
-    parser_submit.add_argument('--entity', '--eid', '--entityId', type=str,
+    parser_submit.add_argument('--entity', '--eid', '--entityId', '--id', type=str,
             help='Synapse ID of the entity to be submitted')
     parser_submit.add_argument('--file', '-f', type=str,
             help='File to be submitted to the challenge')
-    parser_submit.add_argument('--parentId', '--pid', type=str, dest='parentid',
+    parser_submit.add_argument('--parentId', '--parentid', '--parent', type=str, dest='parentid',
```
Comment (larssono) [inline synapseclient/__main__.py]: I don't think parent is necessary to put explicitly as it should already work as as does --pare and --pa etc.

File: synapseclient/client.py [hunk 2]
```diff
@@ -2325,14 +2352,63 @@ def submit(self, evaluation, entity, name=None, teamName=None, silent=False):
         entity_version = entity.get('versionNumber', 1)
         entity_id = entity['id']
 
-        name = entity['name'] if (name is None and 'name' in entity) else name
         submission = {'evaluationId'  : evaluation_id,
                       'entityId'      : entity_id,
                       'name'          : name,
-                      'submitterAlias': teamName,
+                      'submitterAlias': team.name if team else submitterAlias,
```
Comment (kkdang) [inline synapseclient/client.py]: How about keeping submitterAlias in a separate property?

File: synapseclient/client.py [hunk 3]
```diff
@@ -2325,14 +2352,63 @@ def submit(self, evaluation, entity, name=None, teamName=None, silent=False):
...
+            #    'isQuotaFilled': False,
+            #    'isRegistered': True,
+            #    'principalId': 1421212}],
+            #  'teamEligibility': {
+            #   'isEligible': True,
+            #   'isQuotaFilled': False,
+            #   'isRegistered': True},
+            #  'teamId': '3325434'}
+            ## Note that isRegistered may be left out
+            print eligibility
```
Comment (kkdang) [inline synapseclient/client.py]: remove print

File: synapseclient/client.py [hunk 4]
```diff
@@ -2325,14 +2338,92 @@ def submit(self, evaluation, entity, name=None, teamName=None, silent=False):
...
+                    if member.get('hasConflictingSubmission', False):
+                        profile = self.getUserProfile(member['principalId'])
+                        raise SynapseError('Team member {username}({id}) has a conflicting submission'.format(**profile))
+                    if member.get('isQuotaFilled', False):
+                        profile = self.getUserProfile(member['principalId'])
+                        raise SynapseError('Team member {username}({id}) has maxed out his/her quota of submissions'.format(**profile))
+                    raise SynapseError('Team member {username}({id}) is not eligible'.format(**profile))
+
+            # TODO do all contributors have to be eligible in order to submit? Or do we just take those that are eligible?
+            contributors = [{'principalId':em['principalId']} for em in eligibility['membersEligibility']]
```
Comment (cbare) [inline synapseclient/client.py]: Only include eligible contributors. Check if submitter is eligible?

File: tests/integration/test_evaluations.py [hunk 5]
```diff
@@ -60,12 +62,14 @@ def test_evaluations():
         ev = syn.store(ev, createOrUpdate=True)
         assert ev.status == 'OPEN'
 
+        # TODO is "participation" deprecated? Should these be removed?
         # Add the current user as a participant
+        myOwnerId = int(syn.getUserProfile()['ownerId'])
+        syn._allowParticipation(ev, myOwnerId)
         syn.joinEvaluation(ev)
```
Comment (cbare) [inline tests/integration/test_evaluations.py]: delete me

**Pull request merged**